File: lua/sum-of-multiples/sum-of-multiples.lua
local function sum_of_multiples(multiples_array)
    local function to(n)
        local sum = 0

        for i = 2, n - 1 do
            for _, m in ipairs(multiples_array) do
                if i % m == 0 then
                    sum = sum + i
                    break
                end
            end
        end

        return sum
    end

    return {to=to}
end

return sum_of_multiples

File: haskell/sum-of-multiples/src/SumOfMultiples.hs
module SumOfMultiples (sumOfMultiples) where

import Data.List (nub)

sumOfMultiples :: [Integer] -> Integer -> Integer
sumOfMultiples factors limit = (sum . nub . concat) listOfFactorLists
    where listOfFactorLists = map (factorList limit) factors

factorList :: Integer -> Integer -> [Integer]
factorList _ 0 = []
factorList limit factor = [factor, factor * 2 .. limit - 1]
